src/server/generic_family.cc

```diff
@@ -575,18 +575,16 @@ OpStatus OpRestore(const OpArgs& op_args, std::string_view key, std::string_view
   return add_res.status();
 }
 
-bool ScanCb(const OpArgs& op_args, PrimeIterator prime_it, const ScanOpts& opts, string* scratch,
-            StringVec* res) {
+bool ScanCb(const OpArgs& op_args, PrimeIterator prime_it, const ScanOpts& opts, StringVec* res) {
   auto& db_slice = op_args.GetDbSlice();
 
   DbSlice::Iterator it = DbSlice::Iterator::FromPrime(prime_it);
   if (prime_it->second.HasExpire()) {
     it = db_slice.ExpireIfNeeded(op_args.db_cntx, it).it;
+    if (!IsValid(it))
```

Comment:
We do a deep copy within `FromPrime` right ? We can avoid it. ExpireIfNeeded is overloaded and it accepts a (PrimeIterator). If we return false below we avoided an allocation

Reply:
can we pass an invalid iterator to ScanCb? I remember that @adiholden   fixed a bug recently around this area

Reply:
@romange I do not know that's why I asked. Also we can avoid the copy within `FromPrime` if we call `ExpireIfNeeded(const Context& cntx, PrimeIterator it)` right ?

Reply:
we can not pass invalid iterator to scancb